frontend/src/components/RoutesOverview.js

```diff
@@ -22,10 +23,21 @@ export default function RoutesOverview(){
             .catch((error) => console.error(error))
     }
 
-    return(
-        <div>
-            <RouteList routes={routes}/>
+    const deleteRoute = (routeId) => {
+        deleteRouteFromList(routeId).then(() => {
+            setRoutes([
+                ...routes.filter((route) => route.id !== routeId),
+            ])
```

Comment:
over here you don't need the array brackets and the spread-operator (...). 

Just 
.then(() => routes.filter((route) => route.id !== routeId)))
works. 

Maybe add a ".catch(console.log);"  at the end.